Wrappers/Python/test/test_algorithms.py

```diff
@@ -1131,7 +1147,210 @@ def test_SIRT_with_TV_warm_start(self):
             sirt.x.as_array(), ig.allocate(0.25).as_array(), 3)
 
 
-class TestSPDHG(unittest.TestCase):
+class TestSPDHG(CCPiTestClass):
+    def setUp(self):
+        data = dataexample.SIMPLE_PHANTOM_2D.get(size=(20, 20))
+        self.subsets = 10
+        
+        data = dataexample.SIMPLE_PHANTOM_2D.get(size=(16, 16))
+
+        ig = data.geometry
+        ig.voxel_size_x = 0.1
+        ig.voxel_size_y = 0.1
+
+        detectors = ig.shape[0]
+        angles = np.linspace(0, np.pi, 90)
+        ag = AcquisitionGeometry.create_Parallel2D().set_angles(
+            angles, angle_unit='radian').set_panel(detectors, 0.1)
+        # Select device
+        dev = 'cpu'
+
+        Aop = ProjectionOperator(ig, ag, dev)
+
+        sin = Aop.direct(data)
+        partitioned_data = sin.partition(self.subsets, 'sequential')
+        self.A = BlockOperator(
+            *[IdentityOperator(partitioned_data[i].geometry) for i in range(self.subsets)])
+        self.A2 = BlockOperator(
+            *[IdentityOperator(partitioned_data[i].geometry) for i in range(self.subsets)])
+
+        # block function
+        self.F = BlockFunction(*[L2NormSquared(b=partitioned_data[i])
+                                 for i in range(self.subsets)])
+        alpha = 0.025
+        self.G = alpha * FGP_TV()
+
+    def test_SPDHG_defaults_and_setters(self):
+        gamma = 1.
```

Comment:
Can you break this test up, either literally or with comments so it's really easy to find what's failed and why in the future.

Reply:
Done